stdlib/messaging/rabbitmq/src/main/ballerina/src/rabbitmq/Module.md

```diff
@@ -1,99 +1,169 @@
-        io:println("An error occurred while creating the MyQueue1 queue.");
-    }
+It is possible to encrypt the communication between the Ballerina client and the broker by using TLS enabled connections. Client and server authentication (peer verification) is also supported. To enable the TLS support in the RabbitMQ broker, the node has to be [configured](https://www.rabbitmq.com/ssl.html#enabling-tls) to know the location of the Certificate Authority bundle, the server's certificate file, and the server's key. A TLS listener should also be configured to know what port to listen on for TLS-enabled client connections.
 
-    // Declares the queue, MyQueue2.
-    var queueResult2 = newChannel2->queueDeclare({ queueName: "MyQueue2" });
-    if (queueResult2 is rabbitmq:Error) {
-        io:println("An error occurred while creating the MyQueue2 queue.");
-    }
+Connecting to a TLS enabled RabbitMQ node using the Ballerina client can be done by passing a `SecureSocket` record with the appropriate values, to the `ConnectionConfiguration` record when initializing the connection.  
 
-    // Publishing messages to an exchange using a routing key.
-    // Publishes the message using newChannel1 and the routing key named MyQueue1.
-    worker w1 {
-        var sendResult = newChannel1->basicPublish("Hello from Ballerina", "MyQueue1");
-        if (sendResult is rabbitmq:Error) {
-            io:println("An error occurred while sending the message to
-                     MyQueue1 using newChannel1.");
-        }
-    }
+```ballerina
+   rabbitmq:Connection connection = new({ host: "localhost", 
+                                    	    port: 5671, 
+                                            secureSocket: { trustStore: { path: "/path/to/trustStore",
+                                                                        password: "rabbitstore" },
+                                                          keystore: { path: "/path/to/client_key.p12",
+                                                                      password: "MySecretPassword" },
+                                                          verifyHostname: true }});
+```
 
-    // Publishing messages to the same routing key using a different channel.
-    // Publishes the message using newChannel2 and the same routing key named MyQueue1.
-    worker w2 {
-        var sendResult = newChannel2->basicPublish("Hello from Ballerina", "MyQueue1");
-        if (sendResult is rabbitmq:Error) {
-            io:println("An error occurred while sending the message to
-                    MyQueue1 using newChannel2.");
+#### Disconnecting from RabbitMQ
+
+To disconnect, simply close the open channels and the connections: 
+
+```ballerina
+   newChannel.close();
+   newConnection.close();
+```
+> Note: Closing the `Channel` may be considered good practice, but isn’t strictly necessary here; it will be done automatically anyway when the underlying `Connection` is closed. 
```

Comment:
```suggestion
> Note: Closing the `Channel` may be a good practice. However, it isn’t strictly necessary in this case as it will be done automatically when the underlying `Connection` is closed. 
```